File: src/org/webfabric/web/servlet/ContextPath.java
package org.webfabric.web.servlet;

import org.webfabric.io.Path;

import javax.servlet.http.HttpServletRequest;

public class ContextPath implements Path {
    private final String value;

    public ContextPath(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }

    public static ContextPath create(HttpServletRequest request) {
        return new ContextPath(request.getContextPath());
    }

    @Override
    public String toString() {
        return value;
    }
}

File: src/org/webfabric/web/servlet/ContextPath.scala
package org.webfabric.web.servlet

import org.webfabric.io.Path;

import javax.servlet.http.HttpServletRequest;

class ContextPath(v: String) extends Path {
  def value(): String = v

  override def toString = v
}

object ContextPath {
  def apply(request: HttpServletRequest):ContextPath = {
    return new ContextPath(request.getContextPath());
  }
}
